.github/workflows/build.yml

```diff
@@ -97,7 +97,28 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - run: ./scripts/ci-select-xcode.sh 15.2
-      - run: make build-xcframework
+      - run: echo "FRAMEWORKRUNID=$(./scripts/xcframework-generated-run.sh)" >> $GITHUB_ENV
+      
+      - uses: actions/download-artifact@v4
+        if: startsWith(github.ref, 'refs/heads/release/')
+        with:
+          name: xcframeworks.zip
+          path: Carthage/
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          run-id: ${{ env.FRAMEWORKRUNID }}
+      
+      - name: Check XCFramework
+        if: startsWith(github.ref, 'refs/heads/release/')
```

Comment:
If this is a release, we dont compile it, we use the one downloaded from the artifact in the step before.

Reply:
`m`: Please add this vital info as a comment.